### PR TITLE
feat: implemented auth, ui for most, and fixed cors

### PR DESCRIPTION
**Pull request by kriptonian1**
## Description

_Give a summary of the change that you have made_ <br />

Fixes #[ISSUENO]

## Dependencies

_Mention any dependencies/packages used_

## Future Improvements

_Mention any improvements to be done in future related to any file/feature_

## Mentions

_Mention and tag the people_

## Screenshots of relevant screens

_Add screenshots of relevant screens_

## Developer's checklist

- [ ] My PR follows the style guidelines of this project
- [ ] I have performed a self-check on my work

**If changes are made in the code:**

- [ ] I have followed the [coding guidelines](https://google.github.io/styleguide/jsguide.html)
- [ ] My changes in code generate no new warnings
- [ ] My changes are breaking another fix/feature of the project
- [ ] I have added test cases to show that my feature works
- [ ] I have added relevant screenshots in my PR
- [ ] There are no UI/UX issues

### Documentation Update

- [ ] This PR requires an update to the documentation at [docs.keyshade.xyz](https://docs.keyshade.xyz)
- [ ] I have made the necessary updates to the documentation, or no documentation changes are required.

___

### **PR Type**
enhancement, bug_fix

___

### **Description**
- Implemented various UI components like Menubar, Team Table, Context Menu, Dropdown Menu, Command UI, Sheet, Dialog, Profile Settings Page, Project Card, Navbar, Account Details Page, Team Header, Table, OTP Verification Page, Combobox, Main Page Layout, Line Tab, OTP Input, Detailed Project Page, Search Model, Button, Sidebar, Scroll Area, Sidebar Tab, Toaster, Avatar, Popover, Switch, and Settings Layout using libraries like Radix UI, cmdk, and Tanstack React Table.
- Enhanced existing components with new features, improved styles, accessibility, and exported them for external usage.
- Connected components with backend APIs for dynamic data handling and implemented navigation and state management for various user interactions.
- Styled components for consistency across light and dark themes and ensured responsive design.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><details><summary>31 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>menubar.tsx</strong><dd><code>Implement Comprehensive Menubar Component with Radix UI</code>&nbsp; &nbsp; </dd></summary>
<hr>

apps/platform/src/components/ui/menubar.tsx
<li>Implemented a comprehensive menubar component using Radix UI <br>primitives.<br> <li> Added various subcomponents like MenubarTrigger, MenubarItem, <br>MenubarCheckboxItem, etc.<br> <li> Utilized utility functions for class name manipulations and <br>conditional rendering.<br> <li> Exported all subcomponents for external usage.<br>

</details>
    

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/217/files#diff-3a30aac05c259b92c558fb19a754a3315f65f2139cc12e84ec6f852b5506184a">+233/-0</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>index.tsx</strong><dd><code>Add Team Table Component with Sorting and Pagination</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/platform/src/components/teams/teamTable/index.tsx
<li>Created a new team table component using Tanstack React Table.<br> <li> Defined columns and data integration for displaying team payment <br>information.<br> <li> Added UI elements like buttons and dropdown menus for actions on table <br>items.<br> <li> Implemented pagination and sorting functionality.<br>

</details>
    

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/217/files#diff-aed7f921b4006dd809b3ffdf114a5eee8eaaa3e68fe789c8ad99a533fcc81b5c">+327/-0</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>context-menu.tsx</strong><dd><code>Develop Context Menu Component with Radix UI</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/platform/src/components/ui/context-menu.tsx
<li>Developed a context menu component using Radix UI primitives.<br> <li> Included subcomponents like ContextMenuTrigger, ContextMenuItem, and <br>ContextMenuCheckboxItem.<br> <li> Styled components for both light and dark themes.<br> <li> Exported all necessary subcomponents for reuse.<br>

</details>
    

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/217/files#diff-5b3819400a2f8465437d671ccdcf9c58dd0b23a6ee060bcae1a37d66ca44ad79">+197/-0</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>dropdown-menu.tsx</strong><dd><code>Create Dropdown Menu Component Using Radix UI</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/platform/src/components/ui/dropdown-menu.tsx
<li>Created a dropdown menu component with Radix UI primitives.<br> <li> Added subcomponents like DropdownMenuTrigger, DropdownMenuItem, and <br>DropdownMenuCheckboxItem.<br> <li> Configured animations and theming for the dropdown components.<br> <li> Exported all subcomponents for external use.<br>

</details>
    

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/217/files#diff-480a3ebcff41eb7d966109cb063d25dc2db79b64a88ac28125b26012bec1bcac">+197/-0</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>command.tsx</strong><dd><code>Implement Command UI Component with cmdk Library</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/platform/src/components/ui/command.tsx
<li>Implemented a command UI component using cmdk library.<br> <li> Added subcomponents like CommandDialog, CommandInput, and CommandList <br>for modular command inputs.<br> <li> Integrated theme support and provided export of all subcomponents.<br>

</details>
    

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/217/files#diff-f0016d0ae5f5903a2f29a5223ac6793ac18a55fbcb78c92f11a69c7c43c85499">+157/-0</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>sheet.tsx</strong><dd><code>Implement Sheet Component for Modal Dialogues Using Radix UI</code></dd></summary>
<hr>

apps/platform/src/components/ui/sheet.tsx
<li>Implemented a sheet component using Radix UI for modal dialogues.<br> <li> Added subcomponents like SheetContent, SheetHeader, and SheetFooter <br>for structured content.<br> <li> Configured animations and accessibility features.<br> <li> Exported all components for reuse.<br>

</details>
    

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/217/files#diff-6f8a64c3832758d72c5223bad021b088e470000e97ca54b6003d1bde18b05f31">+144/-0</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>page.tsx</strong><dd><code>Create OTP Verification Page for Authentication Flow</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/platform/src/app/auth/otp/page.tsx
<li>Created a new page for OTP verification during authentication.<br> <li> Integrated OTP input handling and validation with backend API.<br> <li> Added UI elements for user interaction and feedback.<br> <li> Implemented navigation based on OTP verification result.<br>

</details>
    

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/217/files#diff-3ae286985cfeac090cabd429156a45eeaae5875dc9d9e23d93549e3b289387ba">+147/-0</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>dialog.tsx</strong><dd><code>Develop Dialog Component Using Radix UI</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/platform/src/components/ui/dialog.tsx
<li>Developed a dialog component using Radix UI primitives.<br> <li> Included subcomponents like DialogContent, DialogTrigger, and <br>DialogClose.<br> <li> Styled the dialog for consistency across themes.<br> <li> Exported all subcomponents for external usage.<br>

</details>
    

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/217/files#diff-bd48cb97fd5a56359f4e24b33876d718617378fa7408c4e30fec0bd49c2b6b13">+123/-0</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>page.tsx</strong><dd><code>Create Profile Settings Page for User Management</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/platform/src/app/(main)/settings/@profile/page.tsx
<li>Created a profile settings page for user profile management.<br> <li> Added input fields for user information like email and username.<br> <li> Integrated API for saving changes to the user profile.<br> <li> Implemented UI feedback for action results.<br>

</details>
    

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/217/files#diff-1a5b40c13739f06516b1663f453d1d70a34e3dac1fb9826974c959be14041b22">+117/-0</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>index.tsx</strong><dd><code>Develop Project Card Component with Context Menu Integration</code></dd></summary>
<hr>

apps/platform/src/components/dashboard/projectCard/index.tsx
<li>Developed a project card component for dashboard views.<br> <li> Integrated context menu for project actions like edit and delete.<br> <li> Displayed project details such as environment and configuration <br>counts.<br> <li> Added navigation to detailed project views.<br>

</details>
    

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/217/files#diff-7f68d914aadd2f2c1e346c7dbaf0e98237164aa6e019a068de5bb2c35474f423">+141/-0</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>index.tsx</strong><dd><code>Enhance Navbar with Search and Account Dropdown Features</code>&nbsp; </dd></summary>
<hr>

apps/platform/src/components/shared/navbar/index.tsx
<li>Updated the navbar component with new features.<br> <li> Added search functionality with a dropdown for account actions.<br> <li> Integrated theme switching and user avatar display.<br> <li> Improved responsive design and accessibility.<br>

</details>
    

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/217/files#diff-1463cdbabc4495c0e3893848022efb23ea972381c8d4de0f1ad206b210a0a6ca">+111/-0</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>page.tsx</strong><dd><code>Create Account Details Completion Page for New Users</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/platform/src/app/auth/account-details/page.tsx
<li>Created a new page for users to complete their account details <br>post-signup.<br> <li> Added input fields for additional user information.<br> <li> Connected the form to the backend for data submission.<br> <li> Implemented navigation based on the completion of the account setup.<br>

</details>
    

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/217/files#diff-7370bcf930b1d30e1be3023f2ed82766817cd88aa794f4559a90d534a736a821">+114/-0</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>index.tsx</strong><dd><code>Implement Team Header Component with Role and Member Addition</code></dd></summary>
<hr>

apps/platform/src/components/teams/teamHeader/index.tsx
<li>Implemented a header component for the team management section.<br> <li> Added buttons for creating new roles and adding team members.<br> <li> Integrated dialogs for inputting new team member and role details.<br>

</details>
    

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/217/files#diff-095f1be3d2e6674cf99426d3a1d5af1443a73385f72ee40b60b69ec9e5bc47af">+125/-0</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>table.tsx</strong><dd><code>Implement Basic Table Component with Modular Subcomponents</code></dd></summary>
<hr>

apps/platform/src/components/ui/table.tsx
<li>Implemented a table component using basic HTML table elements.<br> <li> Added subcomponents like TableHeader, TableRow, and TableCell for <br>modular table construction.<br> <li> Styled the table for a consistent look across themes.<br> <li> Exported all subcomponents for external usage.<br>

</details>
    

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/217/files#diff-2069e21b0e6b1cf57ef44eed6f44bfa85bfb66e5ebf30ff3df47ec4492581942">+123/-0</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>page.tsx</strong><dd><code>Create Teammate Invitation Page for User Onboarding</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/platform/src/app/auth/invite/page.tsx
<li>Created a new page for inviting teammates to the platform.<br> <li> Added input handling for email invitations.<br> <li> Implemented UI elements for user interaction and feedback.<br> <li> Connected the invitation process with the backend API.<br>

</details>
    

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/217/files#diff-9a4c714d548aeedf7ad6964d718341548a5f78dfea00416fafa60d3dbc621dd7">+88/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>page.tsx</strong><dd><code>Create Authentication Page with Email and Third-Party Options</code></dd></summary>
<hr>

apps/platform/src/app/auth/page.tsx
<li>Created a new authentication page for user login and signup.<br> <li> Added input fields for email and integrated third-party authentication <br>options.<br> <li> Connected the authentication process with the backend API.<br> <li> Implemented navigation based on authentication status.<br>

</details>
    

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/217/files#diff-877fb58ece20e228e7c1cb834472fb2ea0357ecfc751fd3d2e29d0f6906632df">+108/-0</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>combobox.tsx</strong><dd><code>Implement Combobox Component with Search Functionality</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/platform/src/components/ui/combobox.tsx
<li>Implemented a combobox component using Radix UI.<br> <li> Added a dropdown for selecting options with search functionality.<br> <li> Styled the combobox for a consistent look across themes.<br> <li> Exported the component for external usage.<br>

</details>
    

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/217/files#diff-6b8df80f22b744039e2b2b9fa7162ddebb2507b2be13311811ba5268a843f865">+111/-0</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>page.tsx</strong><dd><code>Create Main Page Layout with Project Summary and Edit Functionality</code></dd></summary>
<hr>

apps/platform/src/app/(main)/page.tsx
<li>Created a main page layout for the platform.<br> <li> Integrated project card components for displaying project summaries.<br> <li> Added a sheet component for editing project details.<br> <li> Implemented navigation and state management for UI interactions.<br>

</details>
    

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/217/files#diff-d6c7ffe7a9ea578ea4b0a5587f2759a7a1eac9a7c7d6dfae25ddd0bae48ea8e2">+106/-0</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>line-tab.tsx</strong><dd><code>Implement Line Tab Component for Settings Navigation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/platform/src/components/ui/line-tab.tsx
<li>Implemented a line tab component for navigation within settings.<br> <li> Added dynamic tab generation based on provided labels.<br> <li> Integrated URL-based navigation for tab switching.<br> <li> Styled the tabs for a consistent look across themes.<br>

</details>
    

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/217/files#diff-eee4ad38e152b4b999a71b8aeb7ce4f020bd61d309ac36709449c0c92d3b8ed4">+90/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>input-otp.tsx</strong><dd><code>Implement OTP Input Component with Structured Subcomponents</code></dd></summary>
<hr>

apps/platform/src/components/ui/input-otp.tsx
<li>Implemented an OTP input component using custom hooks.<br> <li> Added subcomponents like InputOTPSlot and InputOTPSeparator for <br>structured OTP input.<br> <li> Styled the OTP input for a consistent look across themes.<br> <li> Exported all subcomponents for external usage.<br>

</details>
    

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/217/files#diff-2905a64d2fff6a886f6db34c148dde4386dff9c2b056ae60ffb51cd7b84c32f7">+68/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>page.tsx</strong><dd><code>Create Detailed Project Page with Secret Management</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/platform/src/app/(main)/project/[project]/page.tsx
<li>Created a detailed project page for individual project management.<br> <li> Added input fields for project secrets and configurations.<br> <li> Integrated a dialog component for adding new project keys.<br> <li> Styled the page for a consistent look across themes.<br>

</details>
    

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/217/files#diff-cb1131481064b5fc8b8125ab607c5b5e9f0f906a12b1015267c67e48fb2ee61a">+88/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>index.tsx</strong><dd><code>Develop Search Model Component for Navbar with Command UI</code></dd></summary>
<hr>

apps/platform/src/components/shared/navbar/searchModel/index.tsx
<li>Developed a search model component for the navbar.<br> <li> Integrated command UI for search functionality.<br> <li> Added suggestions and settings options within the search interface.<br> <li> Styled the component for a consistent look across themes.<br>

</details>
    

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/217/files#diff-25f0cc4368d8e7de650937503f9f7e18fe7e0efdc54b36d5d38c2d4f35bc5ba6">+84/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>button.tsx</strong><dd><code>Update Button Component with New Styles and Accessibility Features</code></dd></summary>
<hr>

apps/platform/src/components/ui/button.tsx
<li>Updated the button component with new styles and variants.<br> <li> Added disabled state handling and improved accessibility features.<br> <li> Styled the button for a consistent look across themes.<br> <li> Exported the component for external usage.<br>

</details>
    

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/217/files#diff-06c1ee6f636783ff7b5b835aaf446bf152b8e10933551c430ed15bdb25741dcb">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>index.tsx</strong><dd><code>Implement Sidebar Component with Navigation Links and Combobox</code></dd></summary>
<hr>

apps/platform/src/components/shared/sidebar/index.tsx
<li>Implemented a sidebar component for navigation.<br> <li> Added links to main sections like Dashboard, Teams, and Settings.<br> <li> Integrated a combobox component for additional interactions.<br> <li> Styled the sidebar for a consistent look across themes.<br>

</details>
    

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/217/files#diff-a0228b5ec5c8afa33526f47f35284f2b481d499781118b7b3644677b336f1b1d">+71/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>scroll-area.tsx</strong><dd><code>Implement Scroll Area Component with Custom Scrollbar</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/platform/src/components/ui/scroll-area.tsx
<li>Implemented a scroll area component using Radix UI.<br> <li> Added a custom scrollbar for improved interaction.<br> <li> Styled the scroll area for a consistent look across themes.<br> <li> Exported the component for external usage.<br>

</details>
    

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/217/files#diff-8485ed2ff42ff8e9f14b5c3e7d9eff880414fd491a80d74b5b5f80ba868191d0">+45/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>index.tsx</strong><dd><code>Develop Sidebar Tab Component with Active State Handling</code>&nbsp; </dd></summary>
<hr>

apps/platform/src/components/shared/sidebar/sidebarTab/index.tsx
<li>Developed a sidebar tab component for individual navigation items.<br> <li> Added icons and links for each tab.<br> <li> Integrated active state handling based on the current route.<br> <li> Styled the tabs for a consistent look across themes.<br>

</details>
    

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/217/files#diff-6b9fc80c7780f697c3686a33400a7f81c51a0002e33156afaee06b6294dd9b29">+55/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>sonner.tsx</strong><dd><code>Implement Toaster Component with Theme Support and Configurable </code><br><code>Options</code></dd></summary>
<hr>

apps/platform/src/components/ui/sonner.tsx
<li>Implemented a toaster component using the Sonner library.<br> <li> Integrated theme support for consistent styling.<br> <li> Configured toast options for different types of notifications.<br> <li> Exported the component for external usage.<br>

</details>
    

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/217/files#diff-ee972f454d50d8a76ccc826b3fb97bda23584e44696da9d6341b6c29cc23ab80">+32/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>avatar.tsx</strong><dd><code>Develop Avatar Component with Image and Fallback Support</code>&nbsp; </dd></summary>
<hr>

apps/platform/src/components/ui/avatar.tsx
<li>Developed an avatar component using Radix UI.<br> <li> Added subcomponents like AvatarImage and AvatarFallback for flexible <br>usage.<br> <li> Styled the avatar for a consistent look across themes.<br> <li> Exported all subcomponents for external usage.<br>

</details>
    

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/217/files#diff-2a8ee4c8b436cfb722cdb98c235c59807854583a02a83bd4db20cbae9a7261a2">+47/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>popover.tsx</strong><dd><code>Implement Popover Component with Configurable Positioning and </code><br><code>Animations</code></dd></summary>
<hr>

apps/platform/src/components/ui/popover.tsx
<li>Implemented a popover component using Radix UI.<br> <li> Added a trigger and content area for interactive popovers.<br> <li> Configured animations and positioning for the popover.<br> <li> Exported the component for external usage.<br>

</details>
    

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/217/files#diff-bee3cdbb52c1eec4be3ad0c661968a68adad36ac09ae5af81849d79b378531b1">+28/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>switch.tsx</strong><dd><code>Implement Switch Component with Toggle Functionality</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/platform/src/components/ui/switch.tsx
<li>Implemented a switch component using Radix UI.<br> <li> Added a thumb subcomponent for toggle functionality.<br> <li> Styled the switch for a consistent look across themes.<br> <li> Exported the component for external usage.<br>

</details>
    

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/217/files#diff-b190964f0abba074250206c33ca2e2e1ec6e0b9af3aead97e50688cf72110c6b">+26/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>layout.tsx</strong><dd><code>Create Settings Layout Component with Dynamic Tab Rendering</code></dd></summary>
<hr>

apps/platform/src/app/(main)/settings/layout.tsx
<li>Created a layout component for settings pages.<br> <li> Integrated dynamic rendering based on the selected settings tab.<br> <li> Styled the layout for a consistent look across themes.<br>

</details>
    

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/217/files#diff-5a9c2bfa3ded42faa663b03c07ea79cf31aae88bc15c96f4cc97dba88a377ba4">+29/-0</a>&nbsp; &nbsp; </td>
</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: apps/api/src/main.ts [hunk 1]
```diff
@@ -77,7 +69,7 @@ async function initializeNestApp() {
   app.enableCors({
     credentials: true,
     origin: [
-      'http://localhost:3000',
+      'http://localhost:3025',
```
Comment (rajdip-b) [inline apps/api/src/main.ts:72]: Change this back to 3000 otherwise we will have to alter a lot of docs

File: package.json [hunk 2]
```diff
@@ -91,42 +91,41 @@
...
     "test:web": "turbo run test --filter=web",
-    "test:workspace": "turbo run test --filter=workspace",
-    "db:generate-types": "pnpm run --filter=api db:generate-types",
-    "db:generate-migrations": "pnpm run --filter=api db:generate-migrations",
-    "db:deploy-migrations": "pnpm run --filter=api db:deploy-migrations",
-    "db:validate": "pnpm run --filter=api db:validate",
-    "db:format": "pnpm run --filter=api db:format",
-    "db:reset": "pnpm run --filter=api db:reset",
+    "test:platform": "turbo run test --filter=platform",
+    "db:generate-types": "pnpm dlx prisma generate --schema=apps/api/src/prisma/schema.prisma",
```
Comment (rajdip-b) [inline package.json:119]: Revert these `db:` scripts back to the one we currently have hosted.

**Review by rajdip-b — CHANGES_REQUESTED** [review]

File: apps/api/src/main.ts (hunk 1 above)
Comment (kriptonian1) [reply apps/api/src/main.ts:72]: Default post is never a good idea, it's best to change the docs

Comment (rajdip-b) [reply apps/api/src/main.ts:72]: I think i dont see any docs for this package yet. So maybe it wont harm to put forward this change

File: package.json (hunk 2 above)
Comment (rajdip-b) [reply package.json:119]: @kriptonian1 need these changes

**Review by rajdip-b — APPROVED** [review]

**Pull request merged**